packages/ui-core/shared/src/lib/selectors/date-range-picker/date-range-picker.component.ts

```
@@ -478,116 +491,171 @@ export class DateRangePickerComponent extends TranslationBaseComponent implement
-			if (this.ranges[range]) {
-				const [rangeStartDate, rangeEndDate] = this.ranges[range];
+		const predefinedRanges = this.dateRangePickerDirective.ranges;
+		const formattedStartDate = moment(dateRange.startDate).format('YYYY-MM-DD');
+		const formattedEndDate = moment(dateRange.endDate).format('YYYY-MM-DD');
 
-				// Create new moment objects for formatted dates to avoid mutation
-				const formattedStartDate = moment(dateRange.startDate).clone().format('YYYY-MM-DD');
-				const formattedEndDate = moment(dateRange.endDate).clone().format('YYYY-MM-DD');
+		// Iterate over the predefined ranges to check for a match
+		for (const rangeLabel in predefinedRanges) {
+			if (predefinedRanges.hasOwnProperty(rangeLabel)) {
```

Biome

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)